lib/rex/post/meterpreter/command_mapper.rb

```diff
@@ -97,6 +98,38 @@ def self.get_commands(*extensions)
     commands
   end
 
+
+  # Get the TLV Type symbols that are defined with the value
+  # Potential return values are [], [:TLV_TYPE_A], and [:TLV_TYPE_A, :PACKET_TYPE_B]
+  #
+  # Returning an array is a solution to having multiple TLV types having the same value, as documented here:
+  # https://github.com/rapid7/metasploit-framework/issues/16267
+  #
+  # @param Integer value The value of the TLV type to retrieve the TLV type names for.
+  # @return [Array<Symbol>] An array of symbols of all TLV types that are defined with the value. Can be empty.
+  def self.get_tlv_names(value)
+    return @@cached_tlv_types[value] unless @@cached_tlv_types[value].nil?
+
+    # Default to arrays that contain TLV Types, so that we only deal with one data type
+    @@cached_tlv_types = Hash.new { |h, k| h[k] = [] }
+
+    available_modules = [
+      ::Rex::Post::Meterpreter,
+      *::Rex::Post::Meterpreter::ExtensionMapper.get_extension_klasses
+    ].uniq
+
+    available_modules.each do |mod|
+      mod.constants.each do |const|
+        next unless const.to_s.start_with?('TLV_TYPE_') || const.to_s.start_with?('PACKET_')
+
+        @@cached_tlv_types[mod.const_get(const)].push(const)
+      end
+    end
+
+    return @@cached_tlv_types[value] unless @@cached_tlv_types[value].nil?
```

Comment:
We'll need to convert the `Set.new` to an array
```
    @@cached_tlv_types[value].to_a
```

Will `@@cached_tlv_types[value].nil?`ever be nil if we're defaulting the hash? 👀 

```
cached_tlv_types = Hash.new { |h, k| h[k] = [] }
>> cached_tlv_types['missing_key']
=> []
```